File: 2325-decode-the-message/2325-decode-the-message.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        dictionary = {}
        t = 0
        for i in key:
            if i not in dictionary.keys() and i != ' ':
                dictionary[i] = ord(i) - (97 + t)
                t += 1

        ans = ''
        for j in message:
            if j == ' ':
                ans += j
            else:
                ans += chr(ord(j) - dictionary[j])
        return ans

File: 2325-decode-the-message/2325-decode-the-message.cpp
class Solution {
public:
    string decodeMessage(string key, string message) {
        map<char, int> dictionary;
        int t = 0;
        for (auto i:key) {
            if (dictionary.count(i) == 0 && i != ' ') {
                dictionary[i] = int(i) - (97 + t);
                t += 1;
            }
        }
        string ans;

        for (auto j:message) {
            if (j == ' ') ans += j;
            else ans += char(int(j) - dictionary[j]);
        }
        return ans;
    }
};